fv/test-connection/test-connection.go

```diff
@@ -74,29 +91,59 @@ func main() {
 	}
 }
 
-func tryConnect(ipAddress, port string) error {
+func tryConnect(ipAddress, port string, sourcePort string, protocol string) error {
 
 	err := utils.RunCommand("ip", "r")
 	if err != nil {
 		return err
 	}
 
-	const testMessage = "hello"
+	uid := uuid.NewV4().String()
+	testMessage := "hello," + uid
+
+	// The reuse library implements a version of net.Dialer that can reuse UDP/TCP ports, which we
+	// need in order to make connection retries work.
+	var d reuse.Dialer
+	if protocol == "udp" {
+		localAddr := "0.0.0.0:" + sourcePort
```

Comment:
Could move this before the protocol switch, and then use in TCP branch as well.

Reply:
Yes, think I accidentally Ctrl-Zed that change (which was the point of pulling out that variable).